File: awa.py
import os

def load_dictionary(file_path):
    dictionary = {}
    with open(file_path, 'r') as file:
        for line in file:
            parts = line.strip().split(';')
            if len(parts) == 4:
                word, antonyms, synonyms, definition = parts
                dictionary[word] = {
                    'antonyms': antonyms.split(', '),
                    'synonyms': synonyms.split(', '),
                    'definition': definition
                }
    return dictionary

def main():
    dict_file = 'dict.txt'
    
    if not os.path.exists(dict_file):
        print("Dictionary file not found. Please make sure 'dict.txt' exists.")
        return
    
    dictionary = load_dictionary(dict_file)
    
    print("Welcome to the Automated Word Assistant (AWA)!")
    print("Type 'exit' to quit.")
    
    while True:
        word = input("Enter a word: ").strip()
        if word.lower() == 'exit':
            print("Goodbye!")
            break
        
        entry = dictionary.get(word)
        if entry:
            print(f"Word: {word}")
            print(f"Definition: {entry['definition']}")
            print(f"Synonyms: {', '.join(entry['synonyms'])}")
            print(f"Antonyms: {', '.join(entry['antonyms'])}")
        else:
            print(f"Sorry, the word '{word}' is not in the dictionary.")

if __name__ == "__main__":
    main()

File: awa.js
async function loadDictionary() {
    const response = await fetch('dict.txt');
    const text = await response.text();
    const lines = text.trim().split('\n');
    const dictionary = {};

    lines.forEach(line => {
        const parts = line.split(';');
        if (parts.length === 4) {
            const word = parts[0];
            const antonyms = parts[1].split(', ');
            const synonyms = parts[2].split(', ');
            const definition = parts[3];
            dictionary[word] = {
                antonyms: antonyms,
                synonyms: synonyms,
                definition: definition
            };
        }
    });
    
    return dictionary;
}

async function searchWord() {
    const wordInput = document.getElementById('wordInput').value.trim();
    const resultDiv = document.getElementById('result');

    if (!wordInput) {
        resultDiv.innerHTML = '<p>Please enter a word.</p>';
        return;
    }

    const dictionary = await loadDictionary();
    const entry = dictionary[wordInput];

    if (entry) {
        resultDiv.innerHTML = `
            <h3>${wordInput}</h3>
            <p><strong>Definition:</strong> ${entry.definition}</p>
            <p><strong>Synonyms:</strong> ${entry.synonyms.join(', ')}</p>
            <p><strong>Antonyms:</strong> ${entry.antonyms.join(', ')}</p>
        `;
    } else {
        resultDiv.innerHTML = `<p>Sorry, the word '${wordInput}' is not in the dictionary.</p>`;
    }
}
